ODIN_II/SRC/partial_map.cpp

```diff
@@ -348,23 +347,31 @@ void instantiate_not_logic(nnode_t* node, short mark, netlist_t* /*netlist*/) {
 }
 
 /*---------------------------------------------------------------------------------------------
- * (function: instantiate_buffer )
+ * (function: eliminate_buffer )
  * 	Buffers just pass through signals
+ * 	Returns true if the buffer could be eliminated
  *-------------------------------------------------------------------------------------------*/
-void instantiate_buffer(nnode_t* node, short /*mark*/, netlist_t* /*netlist*/) {
-    int width = node->num_input_pins;
-    int i;
-
+bool eliminate_buffer(nnode_t* node, short, netlist_t*) {
```

Comment:
this should probably remove the buf node, since disconnecting it will make it unreachable from the net.

Reply:
Im not sure how that would work? Line 136 doesn't clean it up so there must be some way of cleaning it up

Reply:
Im pretty sure you are correct though, cause I am getting memory leaks by the looks

Reply:
did you clear the memeory leak?